mobile/src/user/UserPages.tsx

```diff
@@ -0,0 +1,61 @@
+import { FormEvent, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Button, FormControl, TextField } from "@mui/material";
+
+import { useUserRepository, useUser } from "../shared/SharedHooks";
+
+export function SignIn() {
+  const navigate = useNavigate();
+  const userRepository = useUserRepository();
+  const [_, setUser] = useUser();
+
+  const [name, setName] = useState<string>();
+
+  const handleSubmit = async function (e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    if (!name) {
+      return;
+    }
+
+    try {
+      const signedInUser = await userRepository.signIn(name);
+      setUser(signedInUser);
+      navigate("/");
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit} method="POST">
+      <FormControl>
+        <TextField
+          id="filled-basic"
+          label="Username"
+          variant="filled"
+          onChange={({ target: { value } }) => setName(value)}
+          sx={{ mb: 2 }}
+        />
+        <Button type="submit" variant="contained">
+          Sign In / Sign Up
+        </Button>
+      </FormControl>
+    </form>
+  );
+}
+
+export function SignOut() {
```

Comment:
I'm a bit uncomfortable with components that return nothing - because for me, they are hooks. I understand that you need it here to bind to a route. But perhaps you shouldn't use a route in the first place for sign out, and the sign out link should just have an even handler that signs the user out. 

In short, I think you're using routing for event handling, and that's not its purpose.